contracts/argent_account.cairo

```diff
@@ -118,4 +145,69 @@ mod ArgentAccount {
         }
         ecdsa::check_ecdsa_signature(hash, guardian_backup::read(), signature_r, signature_s)
     }
+
+    #[external]
+    fn triggerEscapeSigner() {
+        // only called via execute
+        asserts::assert_only_self();
+
+        // no escape when the guardian is not set
+        assert(guardian::read() != 0, 'argent: guardian required');
+
+        // no escape if there is a guardian escape triggered by the signer in progress
+        let current_timestamp = escape_active_at::read();
+        let current_type = escape_type::read();
+        if current_timestamp != 0 {
+            assert(current_type == ESCAPE_TYPE_SIGNER, 'argent: cannot override escape');
+        };
+
+        // store new escape
+        let block_timestamp = dummy_syscalls::get_block_timestamp();
+        let new_escape_activation = block_timestamp + ESCAPE_SECURITY_PERIOD;
+        escape_active_at::write(new_escape_activation);
+        escape_type::write(ESCAPE_TYPE_SIGNER);
+    // EscapeSignerTriggered(new_escape_activation); Can't call yet
+    }
+
+    #[external]
+    fn cancelEscape() {
+        // only called via execute
+        asserts::assert_only_self();
+
+        // validate there is an active escape
+        let current_escape_timestamp = escape_active_at::read();
+        let current_escape_type = escape_type::read();
+        assert(current_escape_timestamp != 0, 'argent: no active escape"');
+
+        // clear escape
+        escape_active_at::write(0);
+        escape_type::write(0);
+    // EscapeCanceled(); Can't call yet
+    }
+
+    #[external]
+    fn escapeSigner(new_signer: felt, block_timestamp: felt) {
+        // only called via execute
+        asserts::assert_only_self();
+        // no escape when the guardian is not set
+        assert(guardian::read() != 0, 'argent: guardian required');
+
+        let current_escape_timestamp = escape_active_at::read();
+        let current_escape_type = escape_type::read();
```

Comment:
I'd prefer if this _let current_escape_type_ is done after current_escape_timestamp is being asserted twice